File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let (mut max_profit, mut min_price) = (0, 10000);
        for price in prices {
            max_profit = max_profit.max(price - min_price);
            min_price = min_price.min(price);
        }
        
        max_profit
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0,
            min_price = INT_MAX;
        
        for (auto const &p:prices) {
            max_profit = max(max_profit,p - min_price);
            min_price = min(p,min_price);
        }
        
        return max_profit;
    }
};